src/bin/bench.rs

```diff
@@ -84,19 +86,24 @@ Compilation can be customized with the `bench` profile in the manifest.
 ";
 
 pub fn execute(options: Options, config: &Config) -> CliResult {
+    debug!("executing; cmd=cargo-bench; args={:?}",
+           env::args().collect::<Vec<_>>());
+
+    config.configure(options.flag_verbose,
+                     options.flag_quiet,
+                     &options.flag_color,
+                     options.flag_frozen,
+                     options.flag_locked)?;
+
     let root = find_root_manifest_for_wd(options.flag_manifest_path, config.cwd())?;
+    let ws = Workspace::new(&root, config)?;
 
-    let spec = if options.flag_all {
+    let spec = if options.flag_all || ws.is_virtual() {
```

Comment:
Hm, won't this break a use case when you are in the root of the workspace, but want to exercise a particular package?

I am thinking about something like this:

```
ws_root/
  Cargo.toml
  foo/
    Cargo.toml
  bar/
    Cargo.toml
```

```
$ pwd
~/projects/ws_root/
$ cargo test --pkg foo
```

If I am right, and the tests are green, this also means that some test is missing :)